aws_lambda_powertools/metrics/metric.py

```diff
@@ -61,7 +61,7 @@ def add_metric(self, name: str, unit: Union[MetricUnit, str], value: float) -> N
 
 
 @contextmanager
-def single_metric(name: str, unit: MetricUnit, value: float, namespace: Optional[str] = None):
+def single_metric(name: str, unit: MetricUnit, value: float, namespace: Optional[str] = None) -> Iterator[SingleMetric]:
```

Comment:
TIL: An Iterator *could* be problematic if someone tries to iterate it: https://github.com/python/typeshed/issues/2772

What we could do instead is using `Generator[yield_value, NoSendType, NoReturnType]` resulting in `Generator[SingleMetric, None, None]`.

```suggestion
def single_metric(name: str, unit: MetricUnit, value: float, namespace: Optional[str] = None) -> Generator[SingleMetric, None, None]:
```

Reply:
On a practical level, mypy wise, it leads to the same type result:

![image](https://user-images.githubusercontent.com/3340292/145177332-6072dd82-385c-4637-b3dd-15b5b0d5e6e5.png)

![image](https://user-images.githubusercontent.com/3340292/145177291-af03fb76-5629-4e8d-b9ad-3699fcaefd98.png)